web/components/space-homepage/SpaceLandingPage.tsx

```diff
@@ -110,16 +114,79 @@ export function SpaceLandingPage() {
     [profileListingData?.profile_listing]
   );
 
+  
   const filteredProfileListings = useMemo(() => {
     if (searchQuery === "")
       return shuffleProfiles(allProfileListings, getDayOfYear(new Date()));
 
     const searchQueryLower = searchQuery.toLowerCase();
     const fuse = new Fuse(allProfileListings, FUSE_OPTIONS);
-
     return fuse.search(searchQueryLower).map((result) => result.item);
   }, [allProfileListings, searchQuery]);
 
+ 
+  
+  
+
+  const { userData } = useUserData();
+  const [{ data: profileData }] = useAllProfilesOfUserQuery({
+    variables: { user_id: userData?.id ?? "" },
+  });
+
+
+
+  const idsToProfileScores = useMemo(() => {
```

Comment:
Cool, looks like this works! My thoughts on the algorithm:

1. Perhaps add +10 if a profile has a profile pic.
2. Seems strange to rank based on response length. We shouldn't incentivize longer answers for the sake of it. (e.g. Someone can game this by typing "AAAAAAA...". For headline, I'd suggest something like +3 for 1 word, +5 for 3 words or more. For responses, maybe +2 for at least 1 word, and +4 for 10 words or more.

Reply:
Thank you for the great idea!